### PR TITLE
Fetch regions

### PR DESCRIPTION
**Pull request by cuteolaf**
- fetch regions from the coretime chain
- load and save regions names from and to local storage
- get core ownership from `mask`

### CODE REVIEW

File: src/contexts/regions/index.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+          (endBlockHeight - beginBlockHeight) * RELAY_CHAIN_BLOCK_TIME;
+
+        const nPaid = paid ? parseHNString(paid) : undefined;
+        const id = `${parseHNString(begin)}-${core}-${mask.slice(2)}`;
+        const name = localStorage.getItem(`region-${id}`);
+
+        _regions.push({
+          begin: tsBegin,
+          core,
+          mask: decodeMaskString(mask, Math.floor(timeslicePeriod / 8)),
```
Comment (Szegoo) [inline src/contexts/regions/index.tsx:99]: Do we need to decode the mask? Not sure if we are gonna use it for anything else other than for showing the core ownership property of each region. Even if we do I think it is best to show in in a hex format.

I think the simplest way to figure out the core ownership of a region would be to just divide `mask` / `FULL_MASK`.

Comment (cuteolaf) [reply src/contexts/regions/index.tsx:99]: i don't think it's not a simple math.
we should count the number of bits set to 1.

Comment (Szegoo) [reply src/contexts/regions/index.tsx:99]: Why? We only want to know what proportion of the core can this region occupy.

Comment (cuteolaf) [reply src/contexts/regions/index.tsx:99]: yup. the mask 0000ff00 and 00ff00 are essentially the same in terms of the number of available blocks.

Comment (Szegoo) [reply src/contexts/regions/index.tsx:99]: That is not what a core mask represents. Each polkadot core can process a fixed amount of work every 6 seconds. Currently each core is processing a single para block at a time. With Agile Coretime a single core can process blocks from many different parachains. The core mask determines how the parachains that share the same core share its resources. E.g. there might be two parachains assigned to a single where one of them utilize 40% and the other one utilizes 60% of the core’s capacity.

Comment (Szegoo) [reply src/contexts/regions/index.tsx:99]: You can think of all the parachains prior to Agile Coretime as having full core masks(each para has its own core)

File: src/utils/functions.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { ApiPromise } from '@polkadot/api';
+
+import { CoreMask, CoreMaskString, Timestamp } from '@/models';
+
+// decode core mask hex string into a Uint8Array
+export const decodeMaskString = (
```
Comment (Szegoo) [inline src/utils/functions.ts:6]: Can't this be removed now?

**Review by Szegoo — APPROVED** [review]

**Pull request merged**